analyze/analyze.rs

```diff
@@ -518,20 +531,39 @@ impl traits::Emit for Paths {
 
 /// Find all retaining paths for the given items.
 pub fn paths(items: &mut ir::Items, opts: &opt::Paths) -> Result<Box<traits::Emit>, traits::Error> {
-    items.compute_predecessors();
+    if !opts.descending() {
+        items.compute_predecessors();
+    }
 
-    let mut paths = Paths {
-        items: Vec::with_capacity(opts.functions().len()),
-        opts: opts.clone(),
+    let functions: Vec<ir::Id> = match opts.functions().is_empty() {
```

Comment:
Changed this to a Vec<ir::Id> rather than a BTreeSet so that the items could be sorted by size :)